### DISCUSSION TITLE
What's the purpose of using Xfoil in example 07?

### DISCUSSION THREAD

**Discussion by DuncanGWT** [NLRWindSystems/WEIS#252] [Q&A]
I am trying to run example 07_te_flaps and it is attempting to reference Xfoil which I do not have installed, is there a way to bypass using Xfoil and directly reference polars I have predetermined? If not is their any advice on installing Xfoil on a Mac.

**Comment by ptrbortolotti** [comment]
hello, you can't immediately bypass Xfoil when you want to look at flaps. however with some minor changes to the code you can skip the call to Xfoil and provide your polars here https://github.com/WISDEM/WEIS/blob/6f9de4509c159c1e3c517e0d6a4ab8ac59a4909a/weis/control/dac.py#L738


I have not compiled xfoil on my Mac in a while, but I used to follow these instructions

Here are the step-by-step instructions:
1.	Build Orr-Sommerfeld database in ./orrs
% cd orrs
% pwd
% cd src
% edit osmap.f
 
2.	Find the following line in osmap.f (roughly line 100) :
      DATA OSFILE / '/var/local/codes/orrs/osmap.dat' /
 
Take the absolute directory string which is generated by the pwd command above, paste it in front of the osmap.dat filename, e.g.
      DATA OSFILE / '/usr/whatever/orrs/osmap.dat' /
 
GB: If the pathname string is too long it will cause a problem.  You can just put Xfoil in a place where the path is shorter or you can use this example:
      DATA OSFILE
     &/'/afs/athena.mit.edu/course/16/16_d0006/Codes/orrs/osmap_lx.dat'/
 
3.	Build this library
% cd ../bin
 
Edit Makefile_DP and set the compiler flags to be:
 
#========================================
#  Default compiler flags
FC = gfortran
FLG = -O
PLTLIB = -lX11
FTNLIB =
#========================================
#  Uncomment for Intel Fortran Compiler
#FC = ifort
#FLG = -O -r8
#PLTLIB = -L/usr/X11R6/lib -lX11
#FTNLIB =
 
Compile the library:
% make -f Makefile_DP osgen
% make -f Makefile_DP osmap.o
% cd ..
% bin/osgen osmaps_ns.lst
 
4.	Build the plot library
 
% cd plotlib
Edit Makefile and change Line 71 to be:
 
include ./config.make.gfortranDP
 
% make
 
5.	Build the programs
 
% cd ../bin
% make -f Makefile_gfortran xfoil
% make -f Makefile_gfortran pplot
% make -f Makefile_gfortran pxplot
 
You can ignore errors such as:
install: 'xfoil' and '/Users/gbarter/codes/Xfoil/bin/xfoil' are the same file
make: *** [Makefile_gfortran:144: xfoil] Error 1

**Comment by gbarter** [comment]
There are also many, many forks of XFoil that have tried to make it more portable and batch-friendly.  Do a generic search for "xfoil" on Github and some notable hits are:

https://github.com/mdolab/CMPLXFOIL
https://github.com/DARcorporation/xfoil-python
https://github.com/byuflowlab/Xfoil.jl
https://github.com/RobotLocomotion/xfoil
https://github.com/nolankucd/Xfoil-for-Mac
